Tests/ObservableStoreTests/FutureTests.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  FutureTests.swift
+//  
+//
+//  Created by Gordon Brander on 4/18/23.
+//
+
+import XCTest
+import Combine
+@testable import ObservableStore
+
+final class FutureTests: XCTestCase {
+    var cancellables: Set<AnyCancellable> = Set()
+    
+    override func setUp() {
+        // Put setup code here. This method is called before the invocation
+        // of each test method in the class.
+        
+        // Clear cancellables from last test.
+        cancellables = Set()
+    }
+    
+    enum TestServiceError: Error {
+        case interruptedByIntergalacticHighwayProject
+    }
+    
+    actor TestService {
+        func calculateMeaningOfLife() -> Int {
+            return 42
+        }
+        
+        func failToCalculateMeaningOfLife() throws -> Int {
+            throw TestServiceError.interruptedByIntergalacticHighwayProject
+        }
+    }
+    
+    func testFutureAsyncExtension() throws {
+        let service = TestService()
+        
+        let expectation = XCTestExpectation(
+            description: "Future completes successfully"
+        )
+        
+        Future {
+            await service.calculateMeaningOfLife()
+        }
+        .sink(
+            receiveCompletion: { completion in
+                switch completion {
+                case .finished:
+                    expectation.fulfill()
+                }
+            },
+            receiveValue: { value in
+                XCTAssertEqual(value, 42)
+            }
+        )
+        .store(in: &cancellables)
+        
+        wait(for: [expectation], timeout: 0.1)
```

Comment:
FYI, I found that the more tests you run at once in Xcode the higher this timeout value has to be. I noticed that issue in the Subconscious repo which has many more tests etc. but just worth noting.

Reply:
Interesting! I had wondered if this were true. Thanks!